executor/executor_test.go

```diff
@@ -125,3 +125,43 @@ func TestSetOperation(t *testing.T) {
 	rows := result.Rows()
 	require.Len(t, rows, 0)
 }
+
+func TestCompareIssue38361(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("drop database if exists TEST1")
+	tk.MustExec("create database TEST1")
+	tk.MustExec("use TEST1")
+	tk.MustExec("create table t(a datetime, b bigint, c bigint)")
+	tk.MustExec("insert into t values(cast('2023-08-09 00:00:00' as datetime), 20230809, 20231310)")
+
+	tk.MustQuery("select a > 20230809 from t").Check(testkit.Rows("0"))
+	tk.MustQuery("select a = 20230809 from t").Check(testkit.Rows("1"))
+	tk.MustQuery("select a < 20230810 from t").Check(testkit.Rows("1"))
+	tk.MustQuery("select a < 20231310 from t").Check(testkit.Rows("0"))
```

Comment:
This is the case that numeric can't convert to datetime, so use real data type to compare?

Reply:
Yes

Reply:
Add some comments here?